src/Main.tsx

```diff
@@ -2,22 +2,154 @@ import React from 'react';
 import MatchSelector from './components/MatchSelector';
 import StandingsTable from './components/StandingsTable';
 import './Main.css';
+// import { Players } from './types/index'
 
 
-class Main extends React.Component {
+interface MainState {
+    regionToDisplay: string;
+    groupToDisplay: string;
+    players: string[]
+    matches: string[][];
+    results: string;
+}
+
+class Main extends React.Component<{},MainState> {
+    constructor(props: {}) {
+        super(props);
+        const defaultRegion = "APAC";
+        const defaultGroup = "A";
+        const players = selectPlayers(defaultRegion, defaultGroup);
+        this.state = {
+            regionToDisplay: defaultRegion,
+            groupToDisplay: defaultGroup,
+            players: players,
+            matches: selectMatches(players),
+            results: ""
+        }
+    }
+
+    handleClickRegion(region: string) {
+        if (this.state.regionToDisplay !== region) {
+            const players = selectPlayers(region, 'A');
+            this.setState({
+                // reset view
+                groupToDisplay: 'A',
+                results: "",
+                // update info
+                regionToDisplay: region,
+                players: players,
+                matches: selectMatches(players),
+
+            });
+        }
+    }
+
+    handleClickGroup(group: string) {
+        if (this.state.groupToDisplay !== group) {
+            const players = selectPlayers(this.state.regionToDisplay, group);
+            this.setState({ 
+                // reset view
+                results: "",
+                // update info
+                groupToDisplay: group ,
+                players: players,
+                matches: selectMatches(players),
+            });
+        }
+    }
+
+    handleClickMatch(mIndex: number, wIndex: number) {
+        this.setState({
+            results: 
+                this.state.results + 
+                this.state.matches[mIndex][wIndex] + " beat " + 
+                this.state.matches[mIndex][+!wIndex] + ". \n"
+        });
+    }
+
     render() {
         return (
             <div className="standings-simulator">
                 <div className="standings-table">
-                    <StandingsTable />
+                    <StandingsTable 
+                        // in case of undefined
+                        players={this.state.players}
+                        results={this.state.results}
+                        onClickRegion={(region) => this.handleClickRegion(region)}
+                        onClickGroup={(group) => this.handleClickGroup(group)}
```

Comment:
le mieux c'est de ne pas faire de lambda inline. 
tu définis une fonction dans le corps de la classe et tu la rappelle dans le onClick. L'idée c'est de ne pas définir dans lambda dans le react. https://cdb.reacttraining.com/react-inline-functions-and-performance-bdff784f5578?gi=64e68fb4e540 si  tu veux creuser le sujet

Reply:
handleClickRegion = (region: string) => { // blablabla }

    handleClickGroup = (group: string) => { // blablabla }

     render() {
        const {players, results, matches} = this.state;
        return (
            <div className="standings-simulator">
                <div className="standings-table">
                    <StandingsTable 
                        // in case of undefined
                        players={players}
                        results={results}
                        onClickRegion={this.handleClickRegion}
                        onClickGroup={this.handleClickGroup}
                    />


Is this what you mean? 
Je crois je n'ai pas vraiment compris pourquoi